go/test/endtoend/vtgate/foreignkey/fk_test.go

```diff
@@ -156,3 +160,482 @@ func TestUpdateWithFK(t *testing.T) {
+				"insert into fk_t11(id, col) values (1, 7)",
+				"insert into fk_t13(id, col) values (1, 7)",
+			},
+			dmlQuery: "delete from fk_t10 where id = 1",
+			assertionQueries: []string{
+				"select * from fk_t10 order by id",
+				"select * from fk_t11 order by id",
+				"select * from fk_t13 order by id",
+			},
+		}, {
+			name: "Delete success with cascaded cascade foreign keys",
+			dataQueries: []string{
+				"insert into fk_t10(id, col) values (1, 7), (2, 9)",
+				"insert into fk_t11(id, col) values (1, 7)",
+				"insert into fk_t12(id, col) values (1, 7)",
+			},
+			dmlQuery: "delete from fk_t10 where id = 1",
+			assertionQueries: []string{
+				"select * from fk_t10 order by id",
+				"select * from fk_t11 order by id",
+				"select * from fk_t12 order by id",
+			},
+		}, {
+			name: "Delete success with set null to an update cascade foreign key",
+			dataQueries: []string{
+				"insert into fk_t15(id, col) values (1, 7), (2, 9)",
+				"insert into fk_t16(id, col) values (1, 7), (2, 9)",
+				"insert into fk_t17(id, col) values (1, 7)",
+				"insert into fk_t18(id, col) values (1, 7)",
+			},
+			dmlQuery: "delete from fk_t16 where id = 1",
+			assertionQueries: []string{
+				"select * from fk_t15 order by id",
+				"select * from fk_t16 order by id",
+				"select * from fk_t17 order by id",
+				"select * from fk_t18 order by id",
+			},
+		}, {
+			name: "Delete success with cascade to delete with set null to an update set null foreign key",
+			dataQueries: []string{
+				"insert into fk_t15(id, col) values (1, 7), (2, 9)",
+				"insert into fk_t16(id, col) values (1, 7), (2, 9)",
+				"insert into fk_t17(id, col) values (1, 7)",
+				"insert into fk_t19(id, col) values (1, 7)",
+			},
+			dmlQuery: "delete from fk_t15 where id = 1",
+			assertionQueries: []string{
+				"select * from fk_t15 order by id",
+				"select * from fk_t16 order by id",
+				"select * from fk_t17 order by id",
+				"select * from fk_t19 order by id",
+			},
+		}, {
+			name: "Update success with cascade to an update set null to an update cascade foreign key",
+			dataQueries: []string{
+				"insert into fk_t15(id, col) values (1, 7), (2, 9)",
+				"insert into fk_t16(id, col) values (1, 7), (2, 9)",
+				"insert into fk_t17(id, col) values (1, 7)",
+				"insert into fk_t18(id, col) values (1, 7)",
+			},
+			dmlQuery: "update fk_t15 set col = 3 where id = 1",
+			assertionQueries: []string{
+				"select * from fk_t15 order by id",
+				"select * from fk_t16 order by id",
+				"select * from fk_t17 order by id",
+				"select * from fk_t18 order by id",
+			},
+		},
+	}
+
+	for _, tt := range testcases {
+		for _, testSharded := range []bool{false, true} {
+			t.Run(getTestName(tt.name, testSharded), func(t *testing.T) {
+				mcmp, closer := start(t)
+				defer closer()
+				// To test unsharded case in vtgate, we just change the keyspace we use for the test.
+				if !testSharded {
+					_ = utils.Exec(t, mcmp.VtConn, "use `uks`")
+				}
+
```

Comment:
the default database is not reset, so it will continue to use the unsharded keyspace for all the queries that should be sent to `sharded`

Reply:
Every test starts with a call to `start` which creates a new connection and sets the correct keyspace default.

Reply:
Added code to set the keyspace explicitly